ctapipe/tools/muon_reconstruction.py

```diff
@@ -5,121 +5,229 @@
+        help='Pedestal noise rms',
+        default_value=1.1,
     ).tag(config=True)
 
-    classes = t.List([
-        CameraCalibrator, EventSource
-    ])
+    classes = [
+        CameraCalibrator, EventSource, MuonRingFitter, MuonIntensityFitter,
+    ]
 
-    aliases = t.Dict({
+    aliases = {
+        'i': 'EventSource.input_url',
         'input': 'EventSource.input_url',
-        'outfile': 'MuonDisplayerTool.outfile',
-        'display': 'MuonDisplayerTool.display',
-        'max_events': 'EventSource.max_events',
-        'allowed_tels': 'EventSource.allowed_tels',
-    })
+        'o': 'MuonAnalysis.output',
+        'output': 'MuonAnalysis.output',
+        'max-events': 'EventSource.max_events',
+        'allowed-tels': 'EventSource.allowed_tels',
+    }
 
     def setup(self):
-        self.source: EventSource = self.add_component(
-            EventSource.from_config(parent=self)
-        )
-        if self.source.input_url == '':
-            raise ToolConfigurationError("please specify --input <events file>")
+        if self.output is None:
+            raise ToolConfigurationError('You need to provide an --output file')
+
+        self.source = self.add_component(EventSource.from_config(parent=self))
         self.calib = self.add_component(CameraCalibrator(
             parent=self,
             subarray=self.source.subarray
         ))
-        self.writer = self.add_component(HDF5TableWriter(self.outfile, "muons"))
-
-    def start(self):
-
-        numev = 0
-        self.num_muons_found = defaultdict(int)
+        self.ring_fitter = self.add_component(MuonRingFitter(
+            parent=self,
+        ))
+        self.intensity_fitter = self.add_component(MuonIntensityFitter(
+            parent=self,
+            subarray=self.source.subarray,
+        ))
+        self.cleaning = self.add_component(
+            TailcutsImageCleaner(
+                parent=self,
+                subarray=self.source.subarray,
+            )
+        )
+        self.writer = self.add_component(HDF5TableWriter(
+            self.output, "", add_prefix=True
+        ))
+        self.pixels_in_tel_frame = {}
+        self.field_of_view = {}
+        self.min_pixels.attach_subarray(self.source.subarray)
+        self.pedestal.attach_subarray(self.source.subarray)
 
-        for event in tqdm(self.source, desc='detecting muons'):
+        self.tel_type_ids = {
+            tid: hash(str(tel)) for tid, tel in self.source.subarray.tel.items()
+        }
 
-            self.calib(event)
-            muon_evt = analyze_muon_event(event)
+    def start(self):
+        for event in tqdm(self.source, desc='Processing events: '):
+            self.process_array_event(event)
+
+    def process_array_event(self, event):
+        self.calib(event)
+
+        for tel_id, dl1 in event.dl1.tel.items():
+            self.process_telescope_event(event.index, tel_id, dl1)
+
+        self.writer.write('sim/event/subarray', [event.index, event.mc])
```

Comment:
should be `sim/event/subarray/shower` I think, since subarray is a group in the DL1 format, not a table.  But this can be unified later.

Reply:
ah yes, forgot the last name